fs/remote/blob.go

```diff
@@ -67,12 +67,37 @@ type blob struct {
 	fetchedRegionSet   regionSet
 	fetchedRegionSetMu sync.Mutex
 
+	// regions which are currently being fetched
+	beingFetchedRegions     map[region]struct{}
+	beingFetchedRegionsMu   sync.Mutex
+	beingFetchedRegionsCond sync.Cond
```

Comment:
Sorry for the late review. Could we use singleflight? https://pkg.go.dev/golang.org/x/sync/singleflight

Reply:
The change is ready. Still need to benchmark a bunch of images to make sure that there's no regression. Will release from Draft within the next few days.

Reply:
All set.